packages/admin-panel-server/src/routes/EntitiesRoute.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import { Request, Response, NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import { EntityConnection } from '../connections';
+
+export class EntitiesRoute extends Route {
+  private readonly entityConnection: EntityConnection;
+
+  constructor(req: Request, res: Response, next: NextFunction) {
+    super(req, res, next);
+
+    this.entityConnection = new EntityConnection(req.session);
+  }
+
+  async buildResponse() {
+    const { hierarchyName, entityCode } = this.req.params;
+    const { fields, search } = this.req.query;
```

Comment:
I'm thinking we should leave the `fields` query parameter for front end to decide. The reason is in case we want to implement ancestor chain, we will need to request for extra `child_codes` field?